### ISSUE TITLE
Support HTTPS Jolokia requests

### ISSUE THREAD

**Issue by EBallou** [influxdata/telegraf#2012]
Need assistance and/or enhancement to support https for Jolokia queries

### Relevant telegraf.conf:

```
[[inputs.jolokia]]
#   ## This is the context root used to compose the jolokia url
   context = "/jolokia"
#
#   ## This specifies the mode used
#   # mode = "proxy"
#   #
#   ## When in proxy mode this section is used to specify further
#   ## proxy address configurations.
#   ## Remember to change host address to fit your environment.
#   # [inputs.jolokia.proxy]
#   #   host = "127.0.0.1"
#   #   port = "8080"
#
#
#   ## List of servers exposing jolokia read service
  [[inputs.jolokia.servers]]
     name = "my_app_server"
     host = "my_app_server.domain.info"
     port = "443"
     username = "jolokia"
     password = "jolokia_pw"
```

### System info:

Using telegraf 1.0.0.  Jolokia 1.3.5 on Red Hat 5, Centos 7

### Steps to reproduce:

1. configure telegraf.conf as above
2. start service
3. receive error in -test

Error handling response: Post http://jolokia:jolokia_pw@my_app_server.domain.info:443/jolokia: EOF

### Expected behavior:

URL option to switch to https protocol and connection achieved to receive data

## Feature Request

Get Jolokia plugin to execute https connections

### Proposal:

Update jolokia plugin to be like mqtt plugin where there is optional ssl config and use full url to address endpoints like phpfpm

### Current behavior:
HTTP forced

### Desired behavior:
HTTPS Optional and retreives.

### Use case: [Why is this important (helps with prioritizing requests)]
Jolokia still can expose application sensitive information/functions and should be encrypted regardless when passing authentication information to be most secure.

**Comment by EBallou** [comment]
I've written an update that enables https and also incorporates the response/header timeout.  How can I submit the code?

**Comment by sparrc** [comment]
you need to submit a PR to github, you'll need to figure out how to do that on your own using online resources

**Comment by EBallou** [comment]
Ok.. submitted a PR.

**Comment by domainname** [comment]
Hi there,

Any updates regarding this issue?

**Comment by EBallou** [comment]
Jeff-

I coded a fix, but ran into issues getting approval from my company to release the fix.  It seems most of the roadblocks are gone, but now I need to integrate the timeout settings that were since added.  Hopefully, I may be able to request the pull soon.

---------------------------------------------------------------------------
Eric Ballou
phone:  (720) 568-3027
mobile:  (720) 441-4702

**Comment by domainname** [comment]
Hi @EBallou,

Thank you for your prompt reply. The feature I need is the HTTPS support. It appears that telegraf still uses HTTP protocol all the time. Is there any way to instruct it to use HTTPS instead?

**Comment by danielnelson** [comment]
We are hoping to merge #2278 for 1.4, which should address this issue.  I encourage you to try it out and provide and feedback on that issue.

**Comment by domainname** [comment]
Hi @danielnelson,

Glad to see your new Jolokia plugin. I'd love to test-drive it. How do I deploy the new plugin on my Linux machine, which has telegraf installed already? I'm a little bit lost.

**Comment by dylanmei** [comment]
@domainname You may build and deploy it or, if you are running Docker, you can use the `quay.io/nordstrom/telegraf` image.

**Comment by danielnelson** [comment]
You can also grab the latest binary for any pull request from CircleCI: From the PR near the bottom click on "show all checks" and then "details" for the CircleCI line.  This will redirect you to CircleCI, then you can click on the "Artifacts" tab and expand the tree that appears to see the link to `telegraf.gz`.  Maybe we can figure out a way to get a more predictable link.

Here is the [direct link](https://6936-33258973-gh.circle-artifacts.com/0/tmp/circle-artifacts.60nnzYt/telegraf.gz).  The build is amd64 linux.

**Comment by domainname** [comment]
Thank you Dylan & Daniel. I'll download it and give a try.

**Comment by domainname** [comment]
I downloaded it and tried it yesterday. It works well. Thank you guys for your efforts in the new feature!

**Comment by danielnelson** [comment]
Thanks to @dylanmei, #2278 has been merged and will be in the 1.5.0 release.
